hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -143,6 +150,52 @@ public BackgroundTaskQueue getTasks() {
     return queue;
   }
 
+  public List<ContainerData> chooseContainerForBlockDeletion(int count,
+      ContainerDeletionChoosingPolicy deletionPolicy)
+      throws StorageContainerException {
+    Map<Long, ContainerData> containerDataMap =
+        ozoneContainer.getContainerSet().getContainerMap().entrySet().stream()
+            .filter(e -> isDeletionAllowed(e.getValue().getContainerData(),
+                deletionPolicy)).collect(Collectors
+            .toMap(Map.Entry::getKey, e -> e.getValue().getContainerData()));
+    return deletionPolicy
+        .chooseContainerForBlockDeletion(count, containerDataMap);
+  }
+
+  private boolean isDeletionAllowed(ContainerData containerData,
+      ContainerDeletionChoosingPolicy deletionPolicy) {
+    if (!deletionPolicy
+        .isValidContainerType(containerData.getContainerType())) {
+      return false;
+    } else if (!containerData.isClosed()) {
+      return false;
+    } else {
+      if (ozoneContainer.getWriteChannel() instanceof XceiverServerRatis) {
+        try {
+          XceiverServerRatis ratisServer =
+              (XceiverServerRatis) ozoneContainer.getWriteChannel();
+          long minReplicatedIndex = ratisServer.getMinReplicatedIndex(PipelineID
+              .valueOf(UUID.fromString(containerData.getOriginPipelineId())));
```

Comment:
If the pipeline does not exist, it should throw GroupMismatchException here. We can check what exception is thrown once. In that case we need to return true.

Reply:
Addressed in the latest patch.